lib/Enhancer.ts

```diff
@@ -187,26 +188,36 @@ export class Enhancer {
           peopleKnows,
           peopleKnownBy,
           predicates: [ ...predicates ].map(value => DF.namedNode(value)),
-          classes: [ ...classes ].map(value => DF.namedNode(value)),
+          personClasses: [ ...classes ].map(value => DF.namedNode(value)),
         });
       });
     });
   }
 
-  public extractActivities(): Promise<{ posts: RDF.NamedNode[]; comments: RDF.NamedNode[] }> {
-    return new Promise<{ posts: RDF.NamedNode[]; comments: RDF.NamedNode[] }>((resolve, reject) => {
+  public extractActivities(): Promise<{
+    posts: RDF.NamedNode[];
+    comments: RDF.NamedNode[];
+    activityClasses: RDF.NamedNode[];
+  }> {
+    return new Promise<{
+      posts: RDF.NamedNode[];
+      comments: RDF.NamedNode[];
+      activityClasses: RDF.NamedNode[];
+    }>((resolve, reject) => {
       // Prepare RDF terms to compare with
       const termType = this.rdfObjectLoader.createCompactedResource('rdf:type').term;
       const termPost = this.rdfObjectLoader.createCompactedResource('snvoc:Post').term;
       const termComment = this.rdfObjectLoader.createCompactedResource('snvoc:Comment').term;
 
       const posts: RDF.NamedNode[] = [];
       const comments: RDF.NamedNode[] = [];
+      const activityClasses: Set<string> = new Set<string>();
       const stream = rdfParser.parse(fs.createReadStream(this.activitiesPath), { path: this.activitiesPath });
       stream.on('error', reject);
       stream.on('data', (quad: RDF.Quad) => {
         if (quad.subject.termType === 'NamedNode' &&
           quad.predicate.equals(termType)) {
+          activityClasses.add(quad.object.value);
```

Comment:
If I understand correctly, this set will in practise always contain only the `snvoc:Post` and `snvoc:Comment` classes.
If this is so, do we really need this set?

An alternative way of working would be to just return a fixed array containing these to classes, which would avoid the overhead of set processing.

Reply:
You are indeed right, for now it contains only those two activity classes but I thought picking them up from the quads this way was better than hardcoding them directly in case more need to be handled in the future..

Reply:
Let's go for the hard-coded option, since we create the class terms anyways already in `termPost` and `termComment`. Will be less processing overhead.